lib/samples/apply_simple_renderer_to_feature_layer/apply_simple_renderer_to_feature_layer_sample.dart

```diff
@@ -27,69 +27,95 @@ class ApplySimpleRendererToFeatureLayerSample extends StatefulWidget {
 
 class _ApplySimpleRendererToFeatureLayerSampleState
     extends State<ApplySimpleRendererToFeatureLayerSample> {
-  final _mapViewController = ArcGISMapView.createController();
+  // The feature layer that will host the symbolized features
   late FeatureLayer _featureLayer;
```

Comment:
@PaulAllanSturm is this one of those places where we can avoid using `late` by initializing it with a constructor

Reply:
Take a look at the latest change. I moved the instantiation from `onMapViewReady` to the property declaration. This design waits to instantiate the property until the first time it is used.

I still kept the `_ready` state just to make sure the map was all ready before interaction occured.

Reply:
Member variables already have lazy initialization as a built-in part of Dart, so there's no need to do the `() {...}()` trick. It could just be:

Option A
```dart
final _featureLayer = FeatureLayer.withFeatureTable(
  ServiceFeatureTable.withUri(
    Uri.parse(
      'https://services.arcgis.com/V6ZHFr6zdgNZuVG0/arcgis/rest/services/Landscape_Trees/FeatureServer/0'),
  ),
);
```

The other option is

Option B
```dart
  late final FeatureLayer _featureLayer;

  // ...

  void onMapViewReady() {
    final serviceFeatureTable = ServiceFeatureTable.withUri(
      Uri.parse(
        'https://services.arcgis.com/V6ZHFr6zdgNZuVG0/arcgis/rest/services/Landscape_Trees/FeatureServer/0'),
    );
    _featureLayer = FeatureLayer.withFeatureTable(serviceFeatureTable);
  }
```

The competing principles are:

Principle A: `late` is hazardous
Principle B: Samples should read as heavily-commented, top-down step-by-step tutorials

Unfortunately the principles are in conflict, because there's no way to avoid `late` without having a bunch of important stuff crammed up in the initializer, out of line with the flow of the tutorial. So I think we have to go with Principle B / Option B in this case.

The general rule would be: declare member variables by calling their constructor if the call is simple; otherwise, declare it `late` and initialize it in `onMapViewReady()`. Once we get into a constructor nested inside a constructor, it's no longer simple.

Reply:
I concur. I'll move the init back to `onMapViewReady`.